pkg/virt-api/webhooks/validating-webhook/validating-webhook.go

```diff
@@ -22,16 +22,21 @@ package validating_webhook
 import (
 	"net/http"
 
+	k8sfield "k8s.io/apimachinery/pkg/util/validation/field"
 	"kubevirt.io/client-go/kubecli"
 
+	netadmitter "kubevirt.io/kubevirt/pkg/network/admitter"
 	validating_webhooks "kubevirt.io/kubevirt/pkg/util/webhooks/validating-webhooks"
 	"kubevirt.io/kubevirt/pkg/virt-api/webhooks"
 	"kubevirt.io/kubevirt/pkg/virt-api/webhooks/validating-webhook/admitters"
 	virtconfig "kubevirt.io/kubevirt/pkg/virt-config"
 )
 
 func ServeVMICreate(resp http.ResponseWriter, req *http.Request, clusterConfig *virtconfig.ClusterConfig) {
-	validating_webhooks.Serve(resp, req, &admitters.VMICreateAdmitter{ClusterConfig: clusterConfig})
+	validating_webhooks.Serve(resp, req, &admitters.VMICreateAdmitter{
+		ClusterConfig:    clusterConfig,
+		NetworkValidator: netadmitter.NewValidator(k8sfield.NewPath("spec"), clusterConfig),
```

Comment:
WDYT about instantiating the NetworkValidator in the caller of `ServeVMICreate`?
As it would allow us to pass instantiate it once and pass it to all consumers (at least within virt-api).

Reply:
Currently there is one consumer, that is the VMI create addmiter.
I prefer to defer such change until we have other cases that use it to realize better whats best.

In any case, I dont think it will be possible to create the network-validator once, from what I saw each consumer pass different "path" ( `spec` vs. `spec.template.spec`).
It may require mutating the network-validator at the caller, or pass the path to the validation method, not sure which one is worse :sweat_smile: